ipfs/src/error.rs

```diff
@@ -95,3 +95,15 @@ impl From<ServerError> for String {
 		}
 	}
 }
+
+impl ::std::fmt::Display for ServerError {
+	fn fmt(&self, f: &mut ::std::fmt::Formatter) -> ::std::fmt::Result {
+        write!(f, "{:?}", self)
+    }
+}
+
+impl ::std::error::Error for ServerError {
+	fn description(&self) -> &str { "ServerError" }
```

Comment:
The methods of the `Error` trait are deprecated. Instead, it's the implementation of `Display` that should write `ServerError`.

Reply:
Indeed, (and I'm inclined to move errors towards `failure` depending on what happens with the new std error trait) but `std::error::Error` is required when [building a hyper server](https://docs.rs/hyper/0.12.10/hyper/server/struct.Builder.html#method.serve).

Reply:
`Error` itself is not deprecated, just implementing the methods on it.
You still should write `impl Error for ServerError {}`.

Reply:
Ah, my mistake. It's been a while since I've manually implemented it at all. :)

I thought maybe you were referring to potential future deprecation: https://github.com/rust-lang/rfcs/pull/2504.